samples: net: secure_mqtt_sensor_actuator: replace error code with enum val

Lorl0rd opened the pull request:
The error code `0x80` is replaced with its corresponding `mqtt_suback_return_code` enum value (see [enum definition](https://github.com/zephyrproject-rtos/zephyr/blob/f412cc643d91660b7acffacdbb6e18b100fccf7c/include/zephyr/net/mqtt.h#L180) for verification. This seems to make the code more readable, as users of this example do not have to look up, what `0x80` encodes.

rlubos commented:
@Lorl0rd Can you fix the commit message? There's is a char/line limit:
> UC5 Commit title exceeds max length (77>75)
Same applies to the commit message body.

There's also something wrong with commit author, doesn't match the signoff entry? Perhaps it's misconfigured `user.name` and `user.email`git config?

jukkar commented on samples/net/secure_mqtt_sensor_actuator/src/mqtt_client.c:
```diff
@@ -214,7 +214,7 @@ static void mqtt_event_handler(struct mqtt_client *const client, const struct mq
 		break;
```
You do not really need to place full HTTP link in the commit message,
something like this would be ok here.
```
The error code `0x80` is replaced with its corresponding
`mqtt_suback_return_code` enum value. This will make the code
more readable, as users of this example do not have to
look up, what `0x80` encodes.
```

Lorl0rd replied:
Oh, right, changed, thank you.

Lorl0rd commented:
I am sorry for the lint errors. I have force pushed a new (amended) commit with corrected formatting and email address. I hope the the PR is now okay to be accepted. Could not figure out the solution for the first time, that is why there are multiple force pushes. I hope this does not cause any problems. I am very thankful for your help and time spent with my PR.

rlubos commented:
There's still some issue with the mail:
>  682390ce985293addb4dc058bf9d5f354b9300dd: author email (71252892+Lorl0rd@users.noreply.github.com) must be a real email and cannot end in @users.noreply.github.com
682390ce985293addb4dc058bf9d5f354b9300dd: author email (71252892+Lorl0rd@users.noreply.github.com) needs to match one of the signed-off-by entries.

Perhaps you need to amend the commit with `--reset-author` option?

pdgendt commented:
> There's still some issue with the mail:
> 
> > 682390ce985293addb4dc058bf9d5f354b9300dd: author email ([71252892+Lorl0rd@users.noreply.github.com](mailto:71252892+Lorl0rd@users.noreply.github.com)) must be a real email and cannot end in @users.noreply.github.com
> > 682390ce985293addb4dc058bf9d5f354b9300dd: author email ([71252892+Lorl0rd@users.noreply.github.com](mailto:71252892+Lorl0rd@users.noreply.github.com)) needs to match one of the signed-off-by entries.
> 
> Perhaps you need to amend the commit with `--reset-author` option?

You need to have a public e-mail address on Github too

Lorl0rd committed:
samples: net: secure_mqtt: error code to enum

The error code `0x80` is replaced with its
corresponding `mqtt_suback_return_code` enum
value. This will make the code more readable, as
users of this example do not have to look up, what
`0x80` encodes.

Signed-off-by: Pályi Lőrinc <palyilorinc33@gmail.com>

Lorl0rd commented:
Took some time to figure out😅 Thank you for your help with this, there is always something new to learn...

kartben commented:
> Took some time to figure out😅 Thank you for your help with this, there is always something new to learn...

Now you're all set for more future pull requests :)
